src/cargo/util/toml/mod.rs

```diff
@@ -269,9 +269,18 @@ impl<'de, P: Deserialize<'de> + Clone> de::Deserialize<'de> for TomlDependency<P
                 let details: IntermediateDependency<P> = IntermediateDependency::deserialize(mvd)?;
                 if let Some(workspace) = details.workspace {
                     if workspace {
+                        if details.default_features.unwrap_or(false)
+                            || details.default_features2.unwrap_or(false)
+                        {
+                            return Err(de::Error::custom(
+                                "`default-features` cannot be `true` for a workspace dependency",
+                            ));
```

Comment:
Since this was previously accepted, I'm not sure this should be initially an error. What do you think about making it a warning that it may become an error in the future?

Also, I'm a bit confused why `true` would be an error, but `false` is a warning.  Since `default-features=true` is the default, why would there be an error if someone specifies it?